### PR TITLE
Add py27-old test case to buildkite

### PR DESCRIPTION
**Pull request by erikjohnston**

### CODE REVIEW

File: changelog.d/4879.misc [hunk 1]
```diff
@@ -0,0 +1 @@
+Readd test case that runs unit tests against oldest supported dependencies.
```
Comment (richvdh) [inline changelog.d/4879.misc]: reinstate? re-add?

File: tox.ini [hunk 2]
```diff
@@ -82,15 +82,32 @@ deps =
 
     mock
     lxml
+    coverage
+
+extras = all
```
Comment (richvdh) [inline tox.ini]: these changes look spurious: the py27-old env deliberately uses its own pip incantations rather than letting tox do them.

Comment (erikjohnston) [reply tox.ini]: Nyargh, I swear it didn't work without these changes so I added them to try and get them into line but I've removed them and now it works and I'm very sad and confused.

File: tox.ini [hunk 3]
```diff
@@ -82,15 +82,22 @@ deps =
...
     lxml
+    coverage
+
 commands =
     /usr/bin/find "{toxinidir}" -name '*.pyc' -delete
     # Make all greater-thans equals so we test the oldest version of our direct
     # dependencies, but make the pyopenssl 17.0, which can work against an
     # OpenSSL 1.1 compiled cryptography (as older ones don't compile on Travis).
     /bin/sh -c 'python -m synapse.python_dependencies | sed -e "s/>=/==/g" -e "s/psycopg2==2.6//" -e "s/pyopenssl==16.0.0/pyopenssl==17.0.0/" | xargs pip install'
-    # Install Synapse itself. This won't update any libraries.
```
Comment (richvdh) [inline tox.ini]: the comment was your friend

File: tox.ini [hunk 4]
```diff
@@ -82,15 +82,22 @@ deps =
...
+
+    # Add this so that coverage will run on subprocesses
+    /bin/sh -c 'echo "import coverage; coverage.process_startup()" > {envsitepackagesdir}/../sitecustomize.py'
+
     pip install -e .
-    {envbindir}/trial {env:TRIAL_FLAGS:} {posargs:tests} {env:TOXSUFFIX:}
+
+    {envbindir}/coverage run "{envbindir}/trial"  {env:TRIAL_FLAGS:} {posargs:tests} {env:TOXSUFFIX:}
```
Comment (richvdh) [inline tox.ini]: do we actually need to run the coverage checks here? is there any reason to imagine the coverage will be any different to the other runs?

Comment (erikjohnston) [reply tox.ini]: The same could be said for a lot of tests tbh, I'd rather we just consistently did coverage stats than have to figure out which tests do and which don't.

Comment (richvdh) [reply tox.ini]: fair

**Pull request merged**